pkg/dns/aws/dns.go

```diff
@@ -101,13 +101,26 @@ func NewProvider(config Config, operatorReleaseVersion string) (*Provider, error
 	if len(region) == 0 {
 		return nil, fmt.Errorf("region is required")
 	}
+	awsconfig := &aws.Config{}
+	awsconfig.Region = &region
+
+	// TODO: This API will only return hostedzone resources (which are global)
+	// when the region is forced to us-east-1. We don't yet understand why.
+	tagRegion := "us-east-1"
+
+	//If the region is in aws china, cn-north-1 or cn-northwest-1, we should:
+	// 1. hard code route53 api endpoint to https://api.route53.cn, as route53 is not GA in AWS China, and aws sdk didn't have the endpoint.
+	// 2. use the aws china regions to setup tagging api correctly in stead of "use-east-1"
+	if (region == "cn-north-1") || (region == "cn-northwest-1") {
+		endpoint := "https://route53.amazonaws.com.cn"
```

Comment:
The endpoint should be `api.roue53.cn`?

Reply:
both works, this one makes more sense I think.

Reply:
I got this from AWS SA:
```
For the China Route53, there are 2 endpoints,  see below

    1  api.route53.cn

    2 route53.amazonaws.com.cn

Based on the info from service team , both 2 endpoints are workable after GA. But the new endpoint(the 2nd ) is recommended.
```

Reply:
@abhinavdahiya @derekwaynecarr I seem to remember there being prior discussion about cloud provider API endpoints being pushed into the cluster config to be consumed by operators — any thoughts on whether we should be pursing a solution there vs. hard-coding endpoint and region rules here?

Reply:
@ironcladlou I assume you are talking about https://issues.redhat.com/browse/CORS-1271 ? I don't think this will land in short time, could we merge this as a temp solution? as we already hard coded to "us-east-1" for now, it make sense to me we can add china regions here? after that jira is done, maybe we can revisit this?

Reply:
@wanghaoran1988 sorry for the delayed response — I think merging a temporary fix is fine here. I do have a suggestion for a clarifying refactor of the implementation:

```go
tagConfig := aws.NewConfig().WithRegion("us-east-1")
if (region == "cn-north-1") || (region == "cn-northwest-1") {
  tagConfig.Endpoint = "https://route53.amazonaws.com.cn"
  tagConfig.Region = region
}

return &Provider{
  // ...
  tags: resourcegroupstaggingapi.New(sess, tagConfig),
  // ...
}
```

Advantage is that it's less code and also makes it clear that the config is for the tagging API specifically.

Thank you for this contribution! We really appreciate it. Thanks again for your patience.